tests/test_project_generation.py

```diff
@@ -0,0 +1,31 @@
+import pybamm_cookiecutter as m
+import pytest
+
+def test_version() -> None:
+    assert m.__version__
+
+@pytest.fixture
+def custom_template(tmpdir):
+    """
+    Generating a project using the template into a tempdir
+    """
+    template = tmpdir.ensure("cookiecutter-template", dir=True)
+    template.join("cookiecutter.json").write('{"project_name": "pybamm_cookie"}')
+
+    repo_dir = template.ensure("{{cookiecutter.project_name}}", dir=True)
+    repo_dir.join("README.rst").write("{{cookiecutter.project_name}}")
+
+    return template
+
+
+def test_bake_custom_project(cookies, custom_template):
```

Comment:
```suggestion
def test_bake_custom_project(cookies: Cookies, custom_template):
```

Don't want to make the work harder but can we also take care of typing a bit from the start just in case you're familiar with it?